utils/calendrical_calculations/src/persian.rs

```diff
@@ -9,10 +9,27 @@
 
 use crate::helpers::{i64_to_i32, I32CastError, IntegerRoundings};
 use crate::rata_die::RataDie;
+
 /// Lisp code reference: <https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L4720>
 // Book states that the Persian epoch is the date: 3/19/622 and since the Persian Calendar has no year 0, the best choice was to use the Julian function.
 const FIXED_PERSIAN_EPOCH: RataDie = crate::julian::fixed_from_julian(622, 3, 19);
 
+// All these years are not leap, while they are considered leap by the 33-year
+// rule. The year following each of them is leap, but it's considered non-leap
+// by the 33-year rule. This table has been tested to match the modified
+// astronomical algorithm based on the 52.5 degrees east meridian from 1178 AP
+// (an arbitrary date before the Persian calendar was adopted in 1304 AP) to
+// 3000 AP (an arbitrary date far into the future).
+const NON_LEAP_CORRECTION: [i32; 78] = [
+    1502, 1601, 1634, 1667, 1700, 1733, 1766, 1799, 1832, 1865, 1898, 1931, 1964, 1997, 2030, 2059,
+    2063, 2096, 2129, 2158, 2162, 2191, 2195, 2224, 2228, 2257, 2261, 2290, 2294, 2323, 2327, 2356,
+    2360, 2389, 2393, 2422, 2426, 2455, 2459, 2488, 2492, 2521, 2525, 2554, 2558, 2587, 2591, 2620,
+    2624, 2653, 2657, 2686, 2690, 2719, 2723, 2748, 2752, 2756, 2781, 2785, 2789, 2818, 2822, 2847,
+    2851, 2855, 2880, 2884, 2888, 2913, 2917, 2921, 2946, 2950, 2954, 2979, 2983, 2987,
```

Comment:
Observation: it's hard not to notice that more leap year overrides are needed the farther into the future we go.

| Century | # Corrections |
|---|---|
| 1200 AP | 0 |
| 1300 AP | 0 |
| 1400 AP | 0 |
| 1500 AP | 1 |
| 1600 AP | 3 |
| 1700 AP | 4 |
| 1800 AP | 3 |
| 1900 AP | 3 |
| 2000 AP | 4 |
| 2100 AP | 5 |
| 2200 AP | 6 | 
| 2300 AP | 6 |
| 2400 AP | 6 |
| 2500 AP | 6 |
| 2600 AP | 6 |
| 2700 AP | 8 |
| 2800 AP | 8 |
| 2900 AP | 9 |

Is there some pattern to the years that need correcting? Can they be fixed by improving the approximation somehow?

Reply:
> Is there some pattern to the years that need correcting? Can they be fixed by improving the approximation somehow?

I looked into it, but couldn't find a good exact pattern that would also lend itself to conversion to and from “fixed” dates. 

Generally, the more into the future you go, the more chaotic it becomes. Is it because of actual astronomical phenomena? Or is it because of the peculiarities of the approximations made in the astronomical calculations code? I have no idea.

Reply:
This is pretty characteristic of doing a local periodic approximation of complex phenomena. Over time the errors start growing.

Reply:
> This is pretty characteristic of doing a local periodic approximation of complex phenomena. Over time the errors start growing.

Yeah, my question was whether we should consider adding approximations over other periods instead of "patching" the approximation.